app/models/user.rb

```diff
@@ -0,0 +1,75 @@
+class User < ApplicationRecord
+  # Include default devise modules. Others available are:
+  # :confirmable, :lockable, :timeoutable, :trackable and :omniauthable
+  devise :database_authenticatable, :registerable,
+         :recoverable, :rememberable, :validatable,
+         :omniauthable, omniauth_providers: %i[facebook]
+
+  validates :first_name, presence: true, length: { maximum: 20 },
+  format: { with: /[a-z]+/i }
+  validates :last_name, presence: true, length: { maximum: 20 },
+  format: { with: /[a-z]+/i }
+
+  has_many :posts, dependent: :destroy
+  has_many :comments, dependent: :destroy
+  has_many :likes, dependent: :destroy
+
+  has_many :friend_requests_sent, foreign_key: :sender_id,
+          class_name: "FriendRequest", dependent: :destroy
+
+  has_many :friend_requests_received, foreign_key: :receiver_id,
+          class_name: "FriendRequest", dependent: :destroy
+
+
+  has_many :friendships, foreign_key: :adder_id, dependent: :destroy
+  has_many :friendships_2, class_name: "Friendship", foreign_key: :added_id,
+            dependent: :destroy
+
+  has_many :friends, through: :friendships, source: :added
```

Comment:
A suggestion to solve #3:

What can help us is have a associations related to `friendships` and `friendships2` above. In fact, to make them more intuitive, probably we should name those `active_friendships` and `passive_friendships` (remember the naming Michael Hartl used in the Rails Tutorial book chapter on following users?).

Now we can have additional associations as such:

```ruby
has_many :active_friends, through :passive_friendships, source: adder
has_many :passive_friends, through :active_friendships, source: added
```

Then it's easy now to determine `ryto.friends` for example: 

```ruby
ryto.friends = people who added ryto + people ryto.added
```